File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        int dup = -1;
        long long total = accumulate(nums.begin(), nums.end(), 0*1LL);
        for( int i = 0 ; i < n  ; i++){
            int idx = abs(nums[i]) - 1;
            if ( nums[idx] < 0 ){
                dup = idx+1;
                break;
            }
            nums[idx] *= -1;
        }
        
        return {dup, (int) ((n*(n+1))/2-(total-dup))};
    }
};

File: 0645-set-mismatch/0645-set-mismatch.go
func abs(x int) int{
    if x < 0 {
        return -1*x
    }else{
        return x
    }
}

func findErrorNums(nums []int) []int {
    n := len(nums)
    dup := -1
    total := 0
    for _,num := range(nums){
        total += num
    }
    
    for _,num := range(nums){
        idx := abs(num)-1
        if nums[idx] < 0 {
            dup = idx+1
            break
        }
        nums[idx] *= -1
    }
    
    return []int{dup, ((n*(n+1))/2-(total-dup))}
}